techmatters/hrm
# viewContact and viewCases permissions

## Description
This PR adds the following permissions:
- viewContact
- viewCase

These permissions are used in two different ways:
- As middleware
  - GET /cases/:id
- Injected in the search queries through `req.searchPermissions`
  - POST /contacts/search
  - GET /cases/
  - POST /cases/search

~TODO: Unit tests~ ✅

### Checklist

### Related Issues
Fixes https://tech-matters.atlassian.net/browse/CHI-1642
Front-end PR: https://github.com/techmatters/flex-plugins/pull/1295

### Verification steps
- Set `viewContact` permission to `[["isSupervisor"], ["isOwner"]]`
- Verify search endpoints return only permitted cases/contacts
- Check GET /cases/:id returns Forbidden for cases you don't own
